code/src/java/plugin/lsttokens/NaturalattacksLst.java

```diff
@@ -122,186 +122,160 @@ protected ParseResult parseTokenWithSeparator(LoadContext context,
-
-		int numTokens = commaTok.countTokens();
-		if (numTokens < 4)
-		{
-			Logging.errorPrint("Invalid Build of " + "Natural Weapon in "
-					+ getTokenName() + ": " + wpn);
-			return null;
-		}
-
-		String attackName = commaTok.nextToken();
-
-		if (attackName.equalsIgnoreCase(Constants.LST_NONE))
-		{
-			Logging.errorPrint("Attempt to Build 'None' as a "
-					+ "Natural Weapon in " + getTokenName() + ": " + wpn);
-			return null;
-		}
+			String numAttacks = commaTok.nextToken();
+			boolean attacksFixed = !numAttacks.isEmpty()
+					&& numAttacks.charAt(0) == '*';
+			if (attacksFixed)
+			{
+				numAttacks = numAttacks.substring(1);
+			}
+			anEquip.put(ObjectKey.ATTACKS_PROGRESS, !attacksFixed);
+			try
+			{
+				int bonusAttacks = Integer.parseInt(numAttacks) - 1;
+				final BonusObj aBonus = Bonus.newBonus(context, "WEAPON|ATTACKS|"
+						+ bonusAttacks);
 
-		attackName = attackName.intern();
-		Equipment anEquip = new Equipment();
-		anEquip.setName(attackName);
-		anEquip.put(ObjectKey.PARENT, obj);
-		/*
-		 * This really can't be raw equipment... It really never needs to be
-		 * referred to, but this means that duplicates are never being detected
-		 * and resolved... this needs to have a KEY defined, to keep it
-		 * unique... hopefully this is good enough :)
-		 *
-		 * CONSIDER This really isn't that great, because it's String dependent,
-		 * and may not remove identical items... it certainly works, but is ugly
-		 */
-		// anEquip.setKeyName(obj.getClass().getSimpleName() + ","
-		// + obj.getKeyName() + "," + wpn);
-		/*
-		 * Perhaps the construction above should be through context just to
-		 * guarantee uniqueness of the key?? - that's too paranoid
-		 */
+				if (aBonus == null)
+				{
+					return new ParseResult.Fail(getTokenName()
+							+ " was given invalid number of attacks: "
+							+ bonusAttacks, context);
+				}
+				anEquip.addToListFor(ListKey.BONUS, aBonus);
+			}
+			catch (NumberFormatException exc)
+			{
+				return new ParseResult.Fail("Non-numeric value for number of attacks in "
+						+ getTokenName() + ": '" + numAttacks + '\'', context);
+			}
 
-		EquipmentHead equipHead = anEquip.getEquipmentHead(1);
+			equipHead.put(StringKey.DAMAGE, commaTok.nextToken());
 
-		String profType = commaTok.nextToken();
-		if (hasIllegalSeparator('.', profType))
-		{
-			return null;
-		}
-		StringTokenizer dotTok = new StringTokenizer(profType, Constants.DOT);
-		while (dotTok.hasMoreTokens())
-		{
-			Type type = Type.getConstant(dotTok.nextToken());
-			anEquip.addToListFor(ListKey.TYPE, type);
-		}
+			// sage_sam 02 Dec 2002 for Bug #586332
+			// allow hands to be required to equip natural weapons
+			int handsrequired = 0;
+			while (commaTok.hasMoreTokens())
+			{
+				final String hString = commaTok.nextToken();
+				if  (hString.startsWith("SPROP="))
+				{
+					anEquip.addToListFor(ListKey.SPECIAL_PROPERTIES,
+						SpecialProperty.createFromLst(hString.substring(6)));
```

Comment:
extra magic number

Reply:
So what's the real best practice here?  Right now, the "magic" 6 is only 2 lines away from what it's associated with - the hString.startsWith("SPROP=").  To extract that and move it away seems like a disservice to me.  I would think it's MORE likely to cause a problem.  The extreme answer is to extract "SPROP=" and have another constant derive it's length (this guarantees one master set of information that is always consistent)... is that what is really done at this point?

Reply:
Oh sorry, my bad - I read this as an index extraction, ignore me.